### PR TITLE
[proto]: Initial proto files

### PR DESCRIPTION
**Pull request by Pterosaur**
Initial version for protobuf files and its conversion.

Based on: https://github.com/sonic-net/DASH/blob/f0517f92b70d05a3b3d55268bec1795c4434f684/documentation/general/dash-sonic-hld.md

### CODE REVIEW

File: proto/acl_group.proto [hunk 1]
```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
+
+package dash.acl_group;
+
+import "types.proto";
+
+message AclGroup {
+    types.IpVersion ip_version = 1;
+    string guid = 2;
```
Comment (lguohan) [inline proto/acl_group.proto:9]: suggest to define guid specific type, do not use string.

Comment (Pterosaur) [inline proto/acl_group.proto:9]: guid native type

File: proto/acl_rule.proto [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    // allow/deny
+    acl_rule.Action action = 2;
+    // true/false   ; if true, stop processing further rules
+    bool terminating = 3;
+    // Optional
+    // Protocol list. E.g. 6-tcp, 17-udp; if not provided, match on all protocols
+    repeated uint32 protocol = 4;
+    // Optional
+    // list of source tag name, if not provided, match on ANY tag or NO tag.
+    repeated string src_tag = 9;
```
Comment (lguohan) [inline proto/acl_rule.proto:24]: should be integer, i think.

File: proto/route_lpm.proto [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    types.IpAddress overlay_sip = 1;
+    // overlay ipv6 dst ip if routing_type is {servicetunnel}
+    types.IpAddress overlay_dip = 2;
+    // underlay ipv4 src ip if routing_type is {servicetunnel}, this is the ST GW VIP (for ST traffic) or custom VIP
+    types.IpAddress underlay_sip = 3;
+    // underlay ipv4 dst ip to override if routing_type is {servicetunnel}, use dst ip from packet if not specified
+    types.IpAddress underlay_dip = 4;
+}
+
+message Route {
```
Comment (lguohan) [inline proto/route_lpm.proto:26]: route_lpm or route?

File: proto/eni.proto [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+enum State{
+    STATE_DISABLED = 0;
+    STATE_ENABLED = 1;
+}
+
+message Eni {
+    string eni_id = 1;
+    bytes mac_address = 2;
+    // Associated Qos profile
+    string qos = 3;
```
Comment (lguohan) [inline proto/eni.proto:16]: this shoule be integer

Comment (Pterosaur) [reply proto/eni.proto:16]: I don't think so, here should be the qos name associated to Qos profile.
https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md#322-qos
image

File: proto/acl_rule.proto (hunk 2 above)
Comment (Pterosaur) [reply proto/acl_rule.proto:24]: In the client view, the tag is a name string, like xxx_Backend, xxx_ResourceManager.
But there is a TagId, like ab6b2357-624c-4a39-ad4e-000000000000, to each TAG group. What do you think to use the GUID type?

**Review by lguohan — APPROVED** [review]
overall, looks good, we are iterate this later.

**Pull request merged**